broadinstitute/single_cell_portal_core
# Adding correlation plots for two-gene searches (SCP-2348)

SCP-2348.  As shown in demo, this shows a correlation plot whenever a user searches for 2 genes, in addition to the existing dotPlot and Heatmap.  The title of the plot includes the correlation coefficient.  This uses the same functions for loading data from Mongo as the current expression plots, and so I would expect performance to be about the same as a cluster plot, since it has to load 2 genes, but no cluster coordinates.

![Gene_correlation_plot_1_2021-06-02](https://user-images.githubusercontent.com/1334561/120530431-826df400-c3ab-11eb-8191-69c3b79e0e53.png)

![Gene_correlation_plot_2_2021-06-02](https://user-images.githubusercontent.com/1334561/120530432-826df400-c3ab-11eb-9239-35fb01e649dd.png)


TO TEST:
1. load a study with expression data
2. do a search for two genes which exist in the study
3. confirm a 'Correlation' tab appears, with the genes as the x and y axis.
